tests/api.c

```diff
@@ -6627,13 +6651,14 @@ static void test_wolfSSL_dtls_export(void)
           wolfSSL_CTX_use_certificate_file(ctx, cliCertFile, SSL_FILETYPE_PEM));
     AssertIntEQ(WOLFSSL_SUCCESS,
             wolfSSL_CTX_use_PrivateKey_file(ctx, cliKeyFile, SSL_FILETYPE_PEM));
-    tcp_connect(&sockfd, wolfSSLIP, server_args.signal->port, 0, 0, NULL);
+    tcp_connect(&sockfd, wolfSSLIP, server_args.signal->port, 1, 0, NULL);
     AssertNotNull(ssl = wolfSSL_new(ctx));
     AssertIntEQ(wolfSSL_set_fd(ssl, sockfd), WOLFSSL_SUCCESS);
 
     /* store server information connected too */
     XMEMSET(&peerAddr, 0, sizeof(peerAddr));
     peerAddr.sin_family = AF_INET;
+    AssertIntEQ(XINET_PTON(AF_INET, wolfSSLIP, &peerAddr.sin_addr),1);
```

Comment:
This will overflow `peerAddr` on a `AF_INET6` address. You need to change something to accomodate IPv6.

Reply:
You might get away with this due to the `sin_zero` padding but I'm not sure that depending on that is a great idea. For reference the struct on my system
```
struct sockaddr_in
  {
    __SOCKADDR_COMMON (sin_);
    in_port_t sin_port;			/* Port number.  */
    struct in_addr sin_addr;		/* Internet address.  */

    /* Pad to size of `struct sockaddr'.  */
    unsigned char sin_zero[sizeof (struct sockaddr)
			   - __SOCKADDR_COMMON_SIZE
			   - sizeof (in_port_t)
			   - sizeof (struct in_addr)];
  };
```

Reply:
`inet_pton(AF_INET, .., ..)` should not overflow the structure and just fail if provided with an incorrect ipv4 address (or at least it's what I hope for a reasonable lib c implementation).

I adapted the code to work with ipv6 address as well

Reply:
This will work with either IPv4 or IPv6 depending on build time options. Any way to make it selectable at run-time?

Reply:
Yes, `getaddrinfo` looks like the only way to convert a string that may contain either an ipv4 or an ipv6 address.
But the API is more complicated, and `WOLFSSL_IP` is already computed at compile time, so I think it is not worthwhile to overcomplicate things.

(there is also a `struct sockaddr_storage` that can accommodate both ipv4 and ipv6, but you still need to figure out the correct family to use with `inet_pton`)

In my opinion, this is good enough. Do you see any particular issue with this particular test? or any situation in which we can't know whether the address is  ipv6 or ipv4 at build time?

Reply:
On second thought this should be enough for now. If we need run time options then we will add it in the future.